types/src/transaction/mod.rs

```diff
@@ -2084,6 +2084,13 @@ pub trait BlockExecutableTransaction: Sync + Send + Clone + 'static {
 
     /// Size of the user transaction in bytes, 0 otherwise
     fn user_txn_bytes_len(&self) -> usize;
+
+    /// Returns true if self is SignatureVerifiedTransaction kind, internally matching either
+    /// BlockMetadata, BlockMetadataExt, or GenesisTransaction. Such transactions have read-write
+    /// conflicts with other transactions in the block, and benefit from synchronization barrier.
+    /// TODO: get rid of this interface (used by block executor) when signaling the barrier
+    /// can happen directly from aptos-vm.
+    fn is_block_metadata_txn(&self) -> bool;
```

Comment:
nit: The naming can be better. i.e. `is_read_write_conflicting_txn`, or something similar. or even maybe `is_block_metadata_or_genesis_txn`. Here, you name it `is_block_metadata_txn`, but then comment that it can be also GenesisTransaction.

Reply:
true, but that would imply the generality current implementation explicitly avoids actually - we decided to do the least invasive implementation specifically for the first transactions.
re:genesis distinction, I can adjust, but the other pov is that genesis is once and in a way special case of block metadata transaction - maybe it can be implementation detail.